### PR TITLE
Add Interactions GameRules

### LINKED ISSUE #21: disableInteractions gamerule
### Is your feature request related to a problem? Please describe.

When you have a server that runs a lobby with trapdoors or other redstone components it is really annoying when they get opened and closed all the time. 

### Describe the solution you'd like

Add a gamerule to disable player and/or mob interactions on trapdoors, doors, buttons, pressure plates, etc.

### Describe alternatives you've considered

None

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by FirstMegaGame4**
Add `canEntityInteractWithBlock` GameRule
Add `canEntityInteractWithEntity` GameRule
Add `canEntityInteractWithWorld` GameRule
Close #21

### CODE REVIEW

File: src/main/java/net/reikeb/notenoughgamerules/Gamerules.java [hunk 1]
```diff
@@ -42,6 +45,9 @@ public static void setupGamerules() {
         ANVIL_DAMAGE = GameRuleRegistry.register("anvilDamage", GameRules.Category.PLAYER, GameRuleFactory.createBooleanRule(true));
         CAN_COPPER_OXIDE = GameRuleRegistry.register("canCopperOxide", GameRules.Category.UPDATES, GameRuleFactory.createBooleanRule(true));
         CAN_HURT_PET_MOBS = GameRuleRegistry.register("canHurtPetMobs", GameRules.Category.MOBS, GameRuleFactory.createBooleanRule(true));
+        CAN_PLAYER_INTERACT_WITH_BLOCK = GameRuleRegistry.register("canPlayerInteractWithAction", GameRules.Category.PLAYER, GameRuleFactory.createBooleanRule(true));
```
Comment (Max094Reikeb) [inline src/main/java/net/reikeb/notenoughgamerules/Gamerules.java]: Please rename the gamerule ID to `canPlayerInteractWithBlock`

File: src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/AbstractButtonBlockMixin.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.Hand;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.reikeb.notenoughgamerules.Gamerules;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
```
Comment (Max094Reikeb) [inline src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/AbstractButtonBlockMixin.java]: Don't forget to organise imports

File: src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/AbstractPressurePlateBlockMixin.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import net.minecraft.block.AbstractPressurePlateBlock;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.reikeb.notenoughgamerules.Gamerules;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
```
Comment (Max094Reikeb) [inline src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/AbstractPressurePlateBlockMixin.java]: Same here

File: src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/LeverBlockMixin.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.Hand;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.reikeb.notenoughgamerules.Gamerules;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
```
Comment (Max094Reikeb) [inline src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/LeverBlockMixin.java]: Same here

**Review by Max094Reikeb — CHANGES_REQUESTED** [review]

File: src/main/java/net/reikeb/notenoughgamerules/Gamerules.java (hunk 1 above)
Comment (FirstMegaGame4) [reply src/main/java/net/reikeb/notenoughgamerules/Gamerules.java]: oh i forgot that

File: src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/AbstractButtonBlockMixin.java (hunk 2 above)
Comment (FirstMegaGame4) [reply src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/AbstractButtonBlockMixin.java]: yes I will do that later when the pr is ready to be merged

**Comment by FirstMegaGame4** [comment]
Fix #21

File: src/main/resources/assets/not_enough_gamerules/lang/en_us.json [hunk 5]
```diff
@@ -6,6 +6,9 @@
   "gamerule.anvilDamage": "Deal anvil damage",
   "gamerule.canCopperOxide": "Can copper oxide",
   "gamerule.canCopperOxide.description": "Defines if normal copper can oxidize",
+  "gamerule.canEntityInteractWithBlock": "Can Entity Interact With Blocks",
+  "gamerule.canEntityInteractWithEntity": "Can Entity Interact With Another Entities",
+  "gamerule.canEntityInteractWithWorld": "Can Entity Interact With World",
```
Comment (Max094Reikeb) [inline src/main/resources/assets/not_enough_gamerules/lang/en_us.json]: No need to make every work start with a cap
`Can entity interact with blocks` is good enough...

**Review by Max094Reikeb — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/not_enough_gamerules/lang/en_us.json [hunk 6]
```diff
@@ -6,6 +6,9 @@
   "gamerule.anvilDamage": "Deal anvil damage",
   "gamerule.canCopperOxide": "Can copper oxide",
   "gamerule.canCopperOxide.description": "Defines if normal copper can oxidize",
+  "gamerule.canEntityInteractWithBlock": "Can entity interact with blocks",
+  "gamerule.canEntityInteractWithEntity": "Can entity interact with another entities",
```
Comment (Max094Reikeb) [inline src/main/resources/assets/not_enough_gamerules/lang/en_us.json]: Not very English, either "interact with other entities", or "interact with another entity", but can't be both. You need to choose one.

**Review by Max094Reikeb — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/not_enough_gamerules/lang/en_us.json (hunk 6 above)
Comment (FirstMegaGame4) [reply src/main/resources/assets/not_enough_gamerules/lang/en_us.json]: I realized that and was going to make a commit to change that

**Review by Max094Reikeb — APPROVED** [review]

**Pull request merged**